### ISSUE TITLE
[pro] pundit authorization currently enforces to use `current_user` key

### ISSUE THREAD

**Issue by michal-samluk** [rmosolgo/graphql-ruby#2347]
1. Currently pundit enforces context to have `current_user` key. So in case of having key: `visitor` it is not configurable (unless overriding `pundit_policy` and `scope_by_pundit_policy`) 
2. Pundit will not work with `current_user` methods. Only as a hash. 
```
class MyContext 
  def current_user
     self[:visitor]
  end 
end 
```

I think the simplest solution would be to extract `current_user = context[:current_user]` line (from methods `#pundit_policy` and `#scope_by_pundit_policy` into a new method: `current_user`. Then it will be easy to override it and add custom logic for `current_user`.

**Comment by rmosolgo** [comment]
👌 Thanks for the suggestion! I'll add something like that to a new release of GraphQL-Pro and follow up here.

**Comment by rmosolgo** [comment]
I've just released graphql-pro 1.10.6 which supports a `#pundit_user` method on your custom query context class. I chose that approach because otherwise, it might be hard to ensure that `#current_user` was overridden in all the right classes. You can see the doc for it here: https://graphql-ruby.org/authorization/pundit_integration.html#custom-user-lookup

Feel free to give it a try and let me know if you run into any trouble with it!
